Feature/cmd k search

dalbanhi opened the pull request:
## New Features Implemented
![cmdk](https://github.com/dalbanhi/git-note/assets/12940551/190e248c-7464-4a75-93d6-21bdb6c6e486)

- [x] Implemented searchable command palette (see`components/shared/Home/LeftSidebar/CommandPalette/CommandPalette.tsx`) 
- [x] Implemented refined server actions with dummy data (`lib/actions/posts.ts`)


## Fixes
### Structure
- [x] Used consistent importing/exporting of dummy data (see `constants/index.ts`)

dalbanhi committed:
Added basic command palette. Need to add 'all' filter and figure out how to filter posts/potential posts into list. Maybe 'forceMount' on the main filters

dalbanhi committed:
Added cmdk functionality and searching for tags and filtering

dalbanhi committed:
fixed merge conflict

brandonetter commented on components/shared/LeftSidebar/CommandPalette/CommandPalette.tsx:
```diff
@@ -0,0 +1,130 @@
+                </button>
+              </div>
+
+              <Command.List className="flex flex-col bg-myBlack-800">
```
```suggestion
              <Command.List className="flex h-[200px] flex-col overflow-y-scroll bg-myBlack-800">
```

You might a more reasonable time with the design if we set some max height- and force the scroll to happen inside the results instead of inside the entire modal. Play with the height and see what a good value would be.

![image](https://github.com/dalbanhi/git-note/assets/4108484/51b521fd-0492-499e-91b8-278e6057b471)

You could even look into customizing the scrollbar a bit by adding a "customScroll" class here and then adding something into our globals:

```js
.customScroll {
  scrollbar-width: thin;
  scrollbar-color: #4a4e69 #1d2032;
}

```
![image](https://github.com/dalbanhi/git-note/assets/4108484/29491fd4-df77-4966-8dcc-2148f589acbc)


This is just an example- but experiment here to get a quality looking component and try using the design for reference here :)

brandonetter commented on components/shared/LeftSidebar/CommandPalette/CommandPalette.tsx:
```diff
@@ -0,0 +1,130 @@
+                    height={12}
+                  ></Image>
+                  <Command.Input
+                    className="w-full bg-myBlack-700 p-2 text-myWhite-300"
```
```suggestion
                    className="w-full bg-myBlack-700 outline-none p-2 text-myWhite-300"
```
Try something like this to get rid of that jarring outline that's overlapping your icon :)

![image](https://github.com/dalbanhi/git-note/assets/4108484/7db90302-33bb-4c3b-b839-9c0167257e52)

brandonetter commented on components/shared/LeftSidebar/PostAndSearch.tsx:
You can add the `outline-none` class to the input used in here as well to remove the selection outline that overlaps your icon/textbox if you'd like

brandonetter commented on components/shared/LeftSidebar/CommandPalette/CommandPalette.tsx:
```diff
@@ -0,0 +1,130 @@
+                        <Command.Item
+                          key={filter.type}
+                          //use cmdk-item class to style the items in the list
+                          className="flex items-center gap-2 p-2 text-myWhite-300 hover:bg-myBlack-700 hover:text-myWhite-100"
```
```suggestion
                          className="cursor-pointer flex items-center gap-2 p-2 text-myWhite-300 hover:bg-myBlack-700 hover:text-myWhite-100"
```

You may want to try `cursor-pointer` here and anywhere else the user can make a selection here. It's a great ux-indicator to tell the user that this is interactable and performs some action when clicked/pressed.

brandonetter reviewed (changes requested):
This feels like it works really well. Even some additional search filters have been added so this looks good.

Some minor style enhancement ideas to get this looking a little more professional and I think this can merge. Feel free to experiment with height and styling suggestions- anything I include in here is just for an example to get you on the right track 👍 

Good work

dalbanhi committed:
Styling fixes and no results found for cmdk

dalbanhi commented:
Hi @brandonetter , I added the fixes you suggested and also went ahead and added a small "No results found" section to the command palette. Thanks for the feedback!
